### PR TITLE
[BUGFIX beta] add inspect and constructor to list of descriptor exceptions

### PR DESCRIPTION
**Pull request by Dhaulagiri**
In addition to `inspect` I ended up needing to add `constructor` as well to get things working (confirmed locally). Let me know if there is anything else I can do here.

Fixes https://github.com/emberjs/ember.js/issues/16049

### CODE REVIEW

**Comment by rwjblue** [comment]
@Dhaulagiri - I pushed some  additional commits here, can you review?

**Comment by rwjblue** [comment]
Also, FWIW, with these changes your demo repo no longer fails due to the descriptor trap (though it does have a failing tests, but I think that is intentional).

**Comment by rwjblue** [comment]
@Dhaulagiri - I added `nodeType` to the list of items that return `undefined` (jQuery is checking to see if it is an element there).

 Also, it seems like the first one is more app specific and might actually be a _correct_ assertion (e.g. something in the app is doing `this.isFree._dependentKeys`).

**Comment by Dhaulagiri** [comment]
> Also, it seems like the first one is more app specific and might actually be a correct assertion

I'll take a closer look.  It may end up being another Chai thing though as this works fine our app but fails trying to do a deep equal comparison between two arrays with Chai.  I wonder if it's trying to iterate through the keys somehow to do the comparison?  It may take me a bit to find time to dig into that but the other stuff that is here will be a big help for us.

**Comment by rwjblue** [comment]
> I'll take a closer look. It may end up being another Chai thing though as this works fine in our app but fails trying to do a deep equal comparison between two arrays with Chai.

Thanks for digging into it!  We can definitely iterate on what we have here....

> It may take me a bit to find time to dig into that but the other stuff that is here will be a big help for us.

No problem! I'm going to land this now then, so it makes it into the next beta (making it easier to test for others also)...

File: packages/ember-metal/lib/properties.js [hunk 1]
```diff
@@ -112,7 +112,19 @@ if (EMBER_METAL_ES5_GETTERS) {
         get(descriptor, property) {
           if (property === DESCRIPTOR) {
             return descriptor;
-          } else if (property === 'toString' || property == 'valueOf' || (Symbol && property === Symbol.toPrimitive)) {
+          } else if (
+            property === 'prototype' ||
+            property === 'constructor' ||
+            property === 'nodeType'
```
Comment (chancancode) [inline packages/ember-metal/lib/properties.js]: Are these not needed for the other branch (no native `Proxy`) as well?

Comment (rwjblue) [reply packages/ember-metal/lib/properties.js]: Everything returns undefined by default in that branch, we only add assertions for the specific properties that we know about...

**Pull request merged**